cmd/ingester/app/processor/span_processor.go

```diff
@@ -59,6 +60,9 @@ func NewSpanProcessor(params SpanProcessorParams) *KafkaSpanProcessor {
 
 // Process unmarshals and writes a single kafka message
 func (s KafkaSpanProcessor) Process(message Message) error {
+	if message == nil {
+		return errors.New("cannot process a nil message")
```

Comment:
Please add a test for this.

Reply:
I don't like this, it reminds me of Node.js code that is littered with null checks at every level due to the lack of type checking. The contract of the processor requires non-nil message; adding nil checks like this doesn't solve the underlying root cause, it only masks it, without providing any help to fix it.